mailet/icalendar/src/test/java/org/apache/james/transport/mailets/ICalendarParserTest.java

```diff
@@ -0,0 +1,195 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.transport.mailets;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.util.Map;
+
+import javax.mail.BodyPart;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeBodyPart;
+
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMailetConfig;
+import org.apache.mailet.base.test.MimeMessageBuilder;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import net.fortuna.ical4j.model.Calendar;
+
+public class ICalendarParserTest {
+    private static final String DEST_ATTRIBUTE = "destAttribute";
+    private static final String SOURCE_ATTRIBUTE = "sourceAttribute";
+
+    private static final String DEST_CUSTOM_ATTRIBUTE = "ics.dest.attribute";
+    private static final String SOURCE_CUSTOM_ATTRIBUTE = "ics.source.attribute";
+
+    private static final String CONTENT_TRANSFER_ENCODING_VALUE ="8bit";
+
+    private static final String CONTENT_TRANSFER_ENCODING = "Content-Transfer-Encoding";
+    private static final String CONTENT_TYPE = "Content-Type";
+    private static final String TEXT_CALENDAR_CHARSET_UTF_8 = "text/calendar; charset=utf-8";
+
+    private static final String RIGHT_ICAL_VALUE = "BEGIN:VCALENDAR\n" +
```

Comment:
Yep, but we have a task to extract from amqpForward the part that extract the attachment from the mime part. So it's just a matter of putting this new mailet between StripAttachment and ICalendarParser